test/operator-framework-e2e/setup.sh

```diff
@@ -204,8 +210,51 @@ cat <<EOF > "${TMP_ROOT}"/catalog/index.yaml
 }
 EOF
 
-docker build -t "${catalog_img}" -f "${TMP_ROOT}"/catalog.Dockerfile "${TMP_ROOT}"/
-$kind load docker-image "${catalog_img}" --name "${kcluster_name}"
+# Add a .indexignore to make catalogd ignore
+# reading the symlinked ..* files that are created when
+# mounting a ConfigMap
+cat <<EOF > "${TMP_ROOT}"/catalog/.indexignore
+..*
+EOF
+
+kubectl create configmap -n "${namespace}" --from-file="${TMP_ROOT}"/catalog.Dockerfile operator-dev-e2e.dockerfile
+kubectl create configmap -n "${namespace}" --from-file="${TMP_ROOT}"/catalog operator-dev-e2e.build-contents
+
+kubectl apply -f - << EOF
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: kaniko
+  namespace: "${namespace}"
+spec:
+  template:
+    spec:
+      containers:
+      - name: kaniko
+        image: gcr.io/kaniko-project/executor:latest
```

Comment:
Could you please explain why we need to build image from within the cluster? Is it possible to avoid dependency on this? E.g. build locally + kind load?

Reply:
We have to build within the cluster because catalogd now relies on fetching information directly from an image registry. Unfortunately kind loading isn't sufficient because it doesn't load it onto a "real" registry that is accessible by catalogd. I went through various iterations (port-forward then push, followed KinD documentation on using a local registry, etc.) when making the change to catalogd and building and pushing within the cluster using kaniko was the solution that worked best for running the test on both mac and linux. When attempting the alternatives there was always a situation where it worked for mac or linux but not the other.

Reply:
Got it, thanks for the background!